pkg/fargate/delete.go

```diff
@@ -23,6 +23,33 @@ func (c *Client) DeleteProfile(name string, waitForDeletion bool) error {
 	if waitForDeletion {
 		return c.waitForDeletion(name)
 	}
+
+	profiles, err := c.api.ListFargateProfiles(&eks.ListFargateProfilesInput{
+		ClusterName: &c.clusterName,
+	})
+
+	if err != nil {
+		return err
+	}
+
+	//If waitForDeletion is false then the profile might still exist until deletion finishes
+	if len(profiles.FargateProfileNames) == 0 ||
```

Comment:
What are these checks checking for?

Reply:
if there are no fargate profiles left, or the only profile left is the one we are currently deleting then we should delete the unused stack/role